src/Sparrow.Server/Utils/DiskStatsGetter/LinuxDiskStatsGetter.cs

```diff
@@ -0,0 +1,157 @@
+using System.Runtime.Versioning;
+using System.Text;
+using Mono.Unix.Native;
+using Sparrow.Logging;
+
+namespace Sparrow.Server.Utils.DiskStatsGetter;
+
+[SupportedOSPlatform("linux")]
+internal class LinuxDiskStatsGetter : DiskStatsGetter<LinuxDiskStatsRawResult>
+{
+    private static readonly Logger Logger = LoggingSource.Instance.GetLogger<LinuxDiskStatsGetter>("Server");
+
+    public LinuxDiskStatsGetter(TimeSpan minInterval) : base(minInterval)
+    {
+    }
+
+    protected override DiskStatsResult CalculateStats(LinuxDiskStatsRawResult currentInfo, State state)
+    {
+        var diff = (currentInfo.Time - state.Prev.Raw.Time).TotalSeconds;
+        var diskSpaceResult = new DiskStatsResult
+        {
+            IoReadOperations = (currentInfo.IoReadOperations - state.Prev.Raw.IoReadOperations) / diff, 
+            IoWriteOperations = (currentInfo.IoWriteOperations - state.Prev.Raw.IoWriteOperations) / diff, 
+            ReadThroughput = new Size((long)((currentInfo.ReadSectors - state.Prev.Raw.ReadSectors) * 512 / diff), SizeUnit.Bytes), 
+            WriteThroughput = new Size((long)((currentInfo.WriteSectors - state.Prev.Raw.WriteSectors) * 512 / diff), SizeUnit.Bytes), 
+            QueueLength = currentInfo.QueueLength
+        };
+        return diskSpaceResult;
+    }
+
+    protected override LinuxDiskStatsRawResult GetDiskInfo(string path)
+    {
+        try
+        {
+            var (major, minor) = GetDiskMajorMinor(path);
+
+            var statPath = $"/sys/dev/block/{major}:{minor}/stat";
+            using var reader = File.OpenRead(statPath);
+            
+            return ReadParse(reader);
+        }
+        catch (Exception e)
+        {
+            if(Logger.IsInfoEnabled)
+                Logger.Info($"Could not get GetDiskInfo of {path}", e);
+            return null;
+        }
+    }
+        
+    private static (ulong Major, ulong Minor) GetDiskMajorMinor(string path)
+    {
+        if (Syscall.stat(path, out var stats) != 0)
+        {
+            var errno = Stdlib.GetLastError();
+            var errorBuilder = new StringBuilder();
+            Syscall.strerror_r(errno, errorBuilder,1024);
+            errorBuilder.Insert(0, $"Failed to get stat for \"{path}\" : ");
+            throw new InvalidOperationException(errorBuilder.ToString());
+        }
+            
+        var deviceId = (stats.st_mode & FilePermissions.S_IFBLK) == FilePermissions.S_IFBLK
+            ? stats.st_rdev
+            : stats.st_dev;
+                
+        //https://sites.uclouvain.be/SystInfo/usr/include/sys/sysmacros.h.html
+        var major = (deviceId & 0x00000000000fff00u) >> 8;
+        major |= (deviceId & 0xfffff00000000000u) >> 32;
+                
+        var minor = (deviceId & 0x00000000000000ffu);
+        minor |= (deviceId & 0x00000ffffff00000u) >> 12;
+        return (major, minor);
+    }
+        
+    private static LinuxDiskStatsRawResult ReadParse(FileStream fileStream)
```

Comment:
How big is this file going to be? Wouldn't it be easier to read that into a buffer and then use `Utf8Parser.TryRead()` directly?

Reply:
This code is part of an old PR.
In this PR I just moved the code to a new file to distinguish between Windows and Linux implementation.
I can change the code to `File.ReadAllBytes` and parse the byte array.
Anyway, a typical stat content is:
 `  374962        0 19405562   175904   819136        0 48659480  5499080        0   255412  5710940   345276        0 108648072    35956        0        0`

Reply:
I changed the code to read all the file content at once and use `Utf8Parser.TryRead`

Reply:
Consider something like:

```
private static DiskStatsRawResult ReadParse(FileStream fileStream)
{
    int valuesIndex = 0;
    Span<long> values = stackalloc long[24];
    Span<byte> buffer = stackalloc byte[1024];
    var read = ReadAllInto(fileStream,buffer);
    var contents = buffer[0..read];
    var time = DateTime.UtcNow;
    while (contents.IsEmpty == false && valuesIndex < values.Length)
    {
        if (Utf8Parser.TryParse(contents, out long v, out var consumed) == false)
            break;
        contents = contents[consumed..];
        values[valuesIndex++] = v;
    }
    
    /*
     *https://www.kernel.org/doc/Documentation/block/stat.txt
     *https://github.com/sysstat/sysstat/blob/master/iostat.c#L429
     */
    int ioWriteOperationsIndex;
    int readSectorsIndex;
    int writeSectorsIndex;
    long? queueLength = null;
    if (valuesIndex >= 11) {
        /* Device or partition */
        ioWriteOperationsIndex = 4;
        readSectorsIndex = 2;
        writeSectorsIndex = 6;
        queueLength = values[8];
    }
    else if (valuesIndex == 4) {
        /* Partition without extended statistics */
        ioWriteOperationsIndex = 2;
        readSectorsIndex = 1;
        writeSectorsIndex = 3;
    }
    else
    {
        if(Logger.IsInfoEnabled)
            Logger.Info($"The stats file {fileStream.Name} should contain at least 4 values");
        return null;
    }

    return new DiskStatsRawResult
    {
        IoReadOperations = values[0], 
        IoWriteOperations = values[ioWriteOperationsIndex],
        ReadSectors = values[readSectorsIndex],
        WriteSectors = values[writeSectorsIndex],
        QueueLength = queueLength,
        
        Time = time
    };

    int ReadAllInto(Stream stream, Span<byte> remainedToRead)
    {
        int totalRead = 0;
        while (true)
        {
            int read = stream.Read(remainedToRead);
            remainedToRead = remainedToRead[read..];
            if (read == 0)
                return totalRead;
            totalRead += read;
        }
    }
}
```